lib/kitchen/driver/ec2.rb

```diff
@@ -57,107 +69,152 @@ class Ec2 < Kitchen::Driver::SSHBase
       default_config :image_id do |driver|
         driver.default_ami
       end
-      default_config :username do |driver|
-        driver.default_username
-      end
-      default_config :endpoint do |driver|
-        "https://ec2.#{driver[:region]}.amazonaws.com/"
-      end
+      default_config :username, nil
 
       default_config :interface, nil
       default_config :associate_public_ip do |driver|
         driver.default_public_ip_association
       end
-      default_config :ssh_timeout, 1
-      default_config :ssh_retries, 3
 
       required_config :aws_access_key_id
       required_config :aws_secret_access_key
       required_config :aws_ssh_key_id
       required_config :image_id
 
+      def self.validation_warn(driver, old, new)
+        driver.warn "WARN: The driver[#{driver.class.name}] config key `#{old}` is deprecated," +
+          " please use `#{new}`"
```

Comment:
I honestly don't know, but could `new` possibly be construed as `Object.new`? Maybe `old_key`/`new_key` despite the names being longer?